supervisor/misc/filter.py

```diff
@@ -124,9 +127,9 @@ def filter_data(coresys: CoreSys, event: dict, hint: dict) -> dict:
 
     if event.get("request"):
         if event["request"].get("url"):
-            event["request"]["url"] = sanitize_url(event["request"]["url"])
+            event["request"]["url"] = sanitize_url(cast(str, event["request"]["url"]))
 
-        headers = event["request"].get("headers", {})
+        headers = cast(dict, event["request"].get("headers", {}))
```

Comment:
Not entirely related to the scope of this PR, but wouldn't this be better?

```suggestion
        if headers := cast(dict, event["request"].get("headers")):
```

And then indent the following block of `if`.

Also, I wonder about the necessity of `cast` here as well.

Reply:
Yeah we can check here instead of creating an empty dict. I'll change accordingly.

cast is only for static type checking (mypy). Otherwise it is considered an object and static type checking is not happy.

Reply:
Oh right, it's dictionary of `object`s, not `dict`s. Fine by me then!